staging/src/k8s.io/apiserver/pkg/admission/decorator.go

```diff
@@ -0,0 +1,22 @@
+package admission
+
+type Decorator interface {
+	Decorate(handler Interface, name string) Interface
+}
+
+type DecoratorFunc func(handler Interface, name string) Interface
+
+func (d DecoratorFunc) Decorate(handler Interface, name string) Interface {
+	return d(handler, name)
+}
+
+type Decorators []Decorator
+
+func (d Decorators) Decorate(handler Interface, name string) Interface {
```

Comment:
// Decorate applies the decorator in inside-out order, i.e. the first decorator in the slice is first applied to the given handler.